internal/graph/reachableresources.go

```diff
@@ -103,7 +103,7 @@ func (crr *ConcurrentReachableResources) ReachableResources(
 			}
 
 			rsm := subjectIDsToResourcesMap(rewrittenSubjectRelation, req.SubjectIds)
-			drsm := rsm.filterForDispatch(dispatched)
+			drsm := rsm.lockForDispatch()
```

Comment:
The name made me think mutexes were involved. Perhaps `asReadOnly` would be better suited - the underlying structs already follow that naming.